packages/odyssey-react-mui/src/Tile.tsx

```diff
@@ -0,0 +1,147 @@
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { ReactElement, memo, useMemo } from "react";
+
+import { NullElement } from "./NullElement";
+import {
+  Card as MuiCard,
+  CardActions as MuiCardActions,
+  CardActionArea as MuiCardActionArea,
+} from "@mui/material";
+import { Button } from "./Button";
+import { ButtonContext } from "./ButtonContext";
+import { Heading5, Paragraph, Support } from "./Typography";
+import { MoreIcon } from "./icons.generated";
+import { HtmlProps } from "./HtmlProps";
+import styled from "@emotion/styled";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "./OdysseyDesignTokensContext";
+import { MenuButton } from "./MenuButton";
+
+export type TileProps = {
+  description?: string;
+  image?: ReactElement | NullElement; // Icon or image
+  menuItems?: ReactElement;
+  overline?: string;
+  title?: string;
+} & ( // You can't have actions and onClick at the same time
+  | {
+      onClick: () => void;
+      button?: never;
+      menuItems?: never;
+    }
+  | {
+      onClick?: never;
+      button?: ReactElement<typeof Button> | NullElement;
+      menuItems?: ReactElement;
+    }
+) &
+  HtmlProps;
+
+const ImageContainer = styled.div<{
+  odysseyDesignTokens: DesignTokens;
+  hasMenuItems: boolean;
+}>`
+  display: flex;
+  align-items: flex-start;
+  max-height: 4.5714285714rem;
+  margin-block-end: ${(props) => props.odysseyDesignTokens.Spacing5};
+  padding-right: ${(props) =>
+    props.hasMenuItems ? props.odysseyDesignTokens.Spacing5 : 0};
+`;
+
+const MenuButtonContainer = styled.div<{ odysseyDesignTokens: DesignTokens }>`
+  position: absolute;
+  right: ${(props) => props.odysseyDesignTokens.Spacing3};
+  top: ${(props) => props.odysseyDesignTokens.Spacing3};
+`;
+
+const Tile = ({
+  button,
+  description,
+  image,
+  menuItems,
+  onClick,
+  overline,
+  title,
+}: TileProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const cardContent = useMemo(() => {
+    return (
+      <>
+        {image && (
+          <ImageContainer
+            odysseyDesignTokens={odysseyDesignTokens}
+            hasMenuItems={menuItems ? true : false}
+          >
+            {image}
+          </ImageContainer>
+        )}
+
+        {overline && <Support component="div">{overline}</Support>}
+        {title && <Heading5 component="div">{title}</Heading5>}
+        {description && (
+          <Paragraph color="textSecondary">{description}</Paragraph>
+        )}
+
+        {button && (
+          <MuiCardActions>
+            <ButtonContext.Provider value={{ isFullWidth: true }}>
+              {button}
+            </ButtonContext.Provider>
+          </MuiCardActions>
+        )}
+      </>
+    );
+  }, [
+    button,
+    description,
+    image,
+    menuItems,
+    overline,
+    title,
+    odysseyDesignTokens,
+  ]);
+
+  return (
+    <MuiCard className={onClick ? "isClickable" : ""}>
```

Comment:
For my own curiosity, why do we need this classname to optionally be present and with this value?

Reply:
The tile has a hover style, but only when the entire surface is clickable — otherwise, it should maintain the default style when hovered. Unfortunately, there's no other indication in the markup that the card is clickable, so I had to add a class.